mbed_devices/mbed_devices.py

```diff
@@ -3,37 +3,53 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 """API for listing devices."""
-from typing import Iterable
+from typing import Tuple, List
 
 from mbed_targets.exceptions import MbedTargetsError
 
-from mbed_devices._internal.candidate_device import CandidateDevice
-from mbed_devices._internal.detect_candidate_devices import detect_candidate_devices
+from mbed_devices._internal.detect_candidate_devices import CandidateDevice, detect_candidate_devices
 from mbed_devices._internal.resolve_target import resolve_target
 from mbed_devices._internal.exceptions import NoTargetForCandidate
 
+from mbed_targets import MbedTarget
 from mbed_devices.device import Device
 from mbed_devices.exceptions import DeviceLookupFailed
 
 
-def get_connected_devices() -> Iterable[Device]:
-    """Returns Mbed Devices connected to host computer."""
-    devices = []
+def get_connected_devices() -> Tuple[List[Device], List[Device]]:
```

Comment:
Personally, I'm not a fan of returning tuples. You need to remember to deal with two values and discarding via `_` is ugly. It makes sense when language supports pattern matching, but in python it just feels clunky.

I wonder if this wouldn't work better if `get_connected_devices` accepted an argument `include_unidentified` or such and always returned a list.

Reply:
I'm not a fan either but there is not a use case where the unidentified are not returned.

Reply:
If we have a fixed structure would it be clearer to return some kind of `Results` object with two attributes `identified` and `unidentified`? Then we side step the potentially subtle bugs that can be caused by tuple unpacking.

Reply:
That's a good idea, I'll take a look at refactoring.

Reply:
I don't follow. Are you saying we always want to return both? Isn't the cli handling 2 cases differently proving that there's a use case for `with_unindentified=True` and `with_unindentified=False`? 🤔 

I don't feel strongly about this, so 🤷‍♂ .

Reply:
You're right it is, but take a look at the updated code and tell me what you think.

Reply:
💯